src/analysis/complex/upper_half_plane.lean

```diff
@@ -0,0 +1,172 @@
+/-
+Copyright (c) 2021 Alex Kontorovich and Heather Macbeth and Marc Masdeu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Kontorovich, Heather Macbeth, Marc Masdeu
+-/
+
+import linear_algebra.special_linear_group
+import analysis.complex.basic
+import group_theory.group_action.defs
+
+/-!
+# The upper half plane and its automorphisms
+
+This file defines `upper_half_plane` to be the upper half plane in `ℂ`.
+
+We furthermore equip it with the structure of an `SL(2,ℝ)` action by
+fractional linear transformations.
+-/
+
+noncomputable theory
+
+open matrix matrix.special_linear_group
+
+open_locale classical big_operators
+
+local attribute [instance] fintype.card_fin_even
+
+@[ext] structure upper_half_plane :=
+(point : ℂ)
+(im_pos' : 0 < point.im)
+
+localized "notation `ℍ` := upper_half_plane" in upper_half_plane
```

Comment:
This is the same notation as the quaternions, but maybe that's what's normally done. Lean doesn't care, only the mathematicians reading it might.

Reply:
Indeed, both are completely standard.  It'd be nice to put the quaternion notation in a locale too, if it's not in one already.